### ISSUE TITLE
Android prod release - gradle tools 3 error

### ISSUE THREAD

**Issue by kelset** [react-navigation/react-navigation#3097]
This is basically a "reload" of issue https://github.com/react-community/react-navigation/issues/1976 because it became polluted and hard to follow.

### Current Behavior
When using Android Studio 3.0, gradle 4.1 & gradle tools 3.0.0 (build tools 26.0.2) (or higher), creating a production release is impossible because of this error:

`uncompiled PNG file passed as argument. Must be compiled first into a .flat file..
error: failed parsing overlays.`

Which is it's caused by the asset files that gets bundled, like: `node_modules_reactnavigation_src_views_assets_backicon.png` 

It's clearly related to AAPT2, the new version present in [gradle tools 3.0](https://developer.android.com/studio/build/gradle-plugin-3-0-0.html) - but even the [migration guide](https://developer.android.com/studio/build/gradle-plugin-3-0-0-migration.html#aapt2) doesn't seem to offer an explanation for this kind of error. This [SO answer](https://stackoverflow.com/questions/42082352/how-to-use-aapt2-where-is-the-document) seems to provide some more details but, yeah, until better explanations here are some workarounds:

* set `android.enableAapt2=false` in `[YOUR PROJECT]/android/gradle.properties`
* https://github.com/react-community/react-navigation/issues/1976#issuecomment-344836523
* https://github.com/react-community/react-navigation/issues/1976#issuecomment-347165119
* https://github.com/react-community/react-navigation/issues/1976#issuecomment-349582827

### Expected Behavior
That everything gets bundled properly with AAPT2.

### Your Environment

As already stated, Android Studio 3.0, gradle 4.1 & gradle tools 3.0.0 (build tools 26.0.2). Doesn't look like it's related to rn/navigation versions, but still happens with latest:

| software         | version
| ---------------- | -------
| react-navigation | beta.21
| react-native     | 0.51
| node             | 8.9.x
| yarn      | 1.3.2

**Comment by hiddentao** [comment]
As stated in #1976 the following temporary workaround got things moving again for me:

```
// gradle.properties

android.enableAapt2=false
```

I'm using Android Studio 3.0.1 on OS X, Gradle 4.1, Build tools 26.0.2

**Comment by wmonecke** [comment]
Still facing this issue here. Any updates / fix on this?

**Comment by otoinsa** [comment]
For me deleting all images did not help,
setting android.enableAapt2=false and syncing gradle did however.

It's not a fix,
but a workaround. :(

**Comment by adsellor** [comment]
Any updates on this?  I tried to set `android.enableAapt2=false` but it didn't help, as well as cleaning assets from all `drawable-*` folders

**Comment by p-ugulino** [comment]
@adsellor I also did `./gradlew clean` to get it working.

**Comment by brentvatne** [comment]
this is an upstream issue https://github.com/facebook/react-native/issues/16906

**Comment by IgorGanapolsky** [comment]
I too was facing this issue, even after placing `android.enableAapt2=false` in my gradle.properties file.  The only thing that fixed it is manually deleting all the **build** folders in my project, and reassembling.
